### ISSUE TITLE
Mining dump trucks get stuck @ gas station

### ISSUE THREAD

**Issue by lowstrife** [MaFi-Games/Captain-of-Industry-issues#405]
It appears trucks from this one specific mining tower want to get stuck at the gas station. No other trucks from any other mining tower get stuck.

Experienced this in both 0.4.4 and 0.4.5b.

![image](https://user-images.githubusercontent.com/65734844/174932192-9cd82ecb-1ece-4542-aee8-6b05a38b3066.png)
![image](https://user-images.githubusercontent.com/65734844/174935994-77cdfdd5-b5e6-42b2-b0af-cf96aebaf6c6.png)

It appears that they are failing to refuel and getting stuck.

![image](https://user-images.githubusercontent.com/65734844/174936006-665b9f43-3114-4fbd-a4b6-5b80bad32f45.png)

Here is the save file with the trucks currently stuck at the gas station. I'm unsure of how to replicate this. None of my other mines have an issue.

Demolishing and rebuilding the mining control tower *appears* to solve the issue. But it is rather game breaking despite the reasonably easy fix. Can easily cause a new player to death spiral and get frustrated. 

[Gas station dumping stuck.zip](https://github.com/MaFi-Games/Captain-of-Industry-issues/files/8954127/Gas.station.dumping.stuck.zip)

**Comment by lowstrife** [comment]
Can confirm - rebuilding the mining tower fixes the issue.

But it still persists throughout my game. It just randomly happened again to a different one of my mines. I haven't touched or modified any part of this mine in several hours. Haven't touched the vehicles, haven't messed with the assigned routing, haven't changed any gas stations... the only thing I can think of is I've modified small portions of the global dumping-in-the-sea areas. Add a little, subtract a little, but still having hundreds of tiles available to dump. 

![image](https://user-images.githubusercontent.com/65734844/174954923-c1edfb2f-3720-429f-bb00-f2bceeef695a.png)

[Gas station dumping stuck2.zip](https://github.com/MaFi-Games/Captain-of-Industry-issues/files/8954827/Gas.station.dumping.stuck2.zip)

**Comment by lowstrife** [comment]
No matter how many times I demolish mining towers or gas stations or re-assign trucks - it still keeps happening.

One thing I've noticed is that it's always dirt which gets stuck. It's never rock.

![image](https://user-images.githubusercontent.com/65734844/175335839-5262d5d8-61f0-4702-b231-7f4ab9285c61.png)

**Comment by lowstrife** [comment]
I might have figured it out!

I have this mining tower (you can see it exists in all the save files posted here). I eventually want to expand this area to put more farms, and I'm filling in the ocean once level below the main grade, and then I'm going to use this tower to put a layer of dirt ontop of it all so I can put farms there. But I'm not ready for that yet. So I've had the filter enabled, dirt only, for this mining tower. But I just made the location really small.

I wonder if there is some weird buggy behavior relating to this specific mining tower. There are plenty of other places to dump dirt, but I wonder if the mine saw this being the closest one, then got stuck in a feedback loop because this tower which it picked has no available space? 

![image](https://user-images.githubusercontent.com/65734844/175337419-cf4344ef-62cf-4309-8f0c-8a7833f0520f.png)

**Comment by lowstrife** [comment]
Oh I just figured it out. 

It's a bug with the revamped custom exporting route system. 

If I manually set a mining tower to export to a loose storage tower, then anytime that mine digs up dirt or rock... the trucks carrying that material will get stuck. You must also assign the ore mine to a dumping area so the trucks can have a place to dump their material.

This is contradictory to the old system where the trucks would find a place to dump their material irregardless.

I added a dumping area to the export route of this mine and the problem was instantly fixed. All my trucks went from bugged to dumping correctly. 

![image](https://user-images.githubusercontent.com/65734844/175339098-9a887157-b49e-49df-96f0-7241c37693c9.png)

Expected behavior: even if I don't have a export route assigned, my mines should still be able to get rid of rock\dirt if there is no specifically assigned route for those items.
